google/cloud/internal/external_account_token_source_aws.cc

```diff
@@ -304,16 +320,35 @@ nlohmann::json ComputeSubjectToken(
 
   auto const host = absl::StrCat("sts.", region, ".amazonaws.com");
   auto const timestamp = internal::FormatV4SignedUrlTimestamp(now);
-  auto const signed_headers = std::string{"host;x-amz-date"};
+  auto const signed_headers_full = [&]() {
+    auto headers = std::vector<std::pair<std::string, std::string>>{
+        {"host", host},
+        {"x-amz-date", timestamp},
+        {"x-goog-cloud-target-resource", target},
+    };
+    // The session token may be empty, in which case we do not need to include
+    // it.
+    if (!secrets.session_token.empty()) {
+      headers.emplace_back("x-amz-security-token", secrets.session_token);
+    }
+    std::sort(headers.begin(), headers.end());
```

Comment:
Q: Is this required?

Reply:
Yes:

https://docs.aws.amazon.com/general/latest/gr/create-signed-request.html#:~:text=use%20lowercase%20characters%2C-,must%20appear%20in%20alphabetical%20order,-%2C%20and%20must%20be

Arguably we could insert these in order ourselves, but I decided to be lazy and more explicit than not in this code.